src/main/java/uk/gov/laa/gpfd/controller/ReportsController.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.laa.gpfd.controller;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.annotation.RegisteredOAuth2AuthorizedClient;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+import uk.gov.laa.gpfd.api.ReportApi;
+import uk.gov.laa.gpfd.api.ReportsApi;
+import uk.gov.laa.gpfd.model.ReportIdGet200Response;
+import uk.gov.laa.gpfd.model.ReportsGet200Response;
+import uk.gov.laa.gpfd.services.MappingTableService;
+import uk.gov.laa.gpfd.services.ReportService;
+import uk.gov.laa.gpfd.services.ReportTrackingTableService;
+
+import java.util.Optional;
+
+@RestController
+@Slf4j
+@RequiredArgsConstructor
+public class ReportsController implements ReportApi, ReportsApi {
+
+    private final ReportTrackingTableService reportTrackingTableService;
+    private final MappingTableService mappingTableService;
+    private final ReportService reportService;
+
+    @Override
+    public Optional<NativeWebRequest> getRequest() {
+        return ReportApi.super.getRequest();
+    }
+
+    @Override
+    public ResponseEntity<ReportIdGet200Response> reportIdGet(Integer id) {
```

Comment:
Do we need to use these names due to the generated code? 

If not, I think it could be more descriptive as like getReportById, getAllReports, etc.

Reply:
Thank you for the suggestion! Unfortunately, this is how the plugin generates the endpoints and models by default, which limits our ability to use more descriptive naming. It’s one of the trade-offs of using this tool. I completely agree that clearer names would improve readability, and I’ll explore whether there’s any flexibility to customize this in future iterations.